src/catch2/internal/catch_run_context.cpp

```diff
@@ -621,6 +625,7 @@ namespace Catch {
             // considered "OK"
             reaction.shouldSkip = true;
         }
+        resetAssertionInfo();
```

Comment:
Same note here, there is `assertionEnded` call few lines up, and the intervening code is relatively trivial.